packages/server/graphql/queries/helpers/repoIntegrationHelpers.ts

```diff
@@ -1,49 +1,33 @@
 import ms from 'ms'
-import JiraProjectKeyId from 'parabol-client/shared/gqlIds/JiraProjectKeyId'
 import {Unpromise} from 'parabol-client/types/generics'
+import IntegrationRepoId from '../../../../client/shared/gqlIds/IntegrationRepoId'
+import JiraProjectKeyId from '../../../../client/shared/gqlIds/JiraProjectKeyId'
 import getRethink from '../../../database/rethinkDriver'
 import {RValue} from '../../../database/stricterR'
+import TaskIntegrationAzureDevOps from '../../../database/types/TaskIntegrationAzureDevOps'
+import TaskIntegrationGitHub from '../../../database/types/TaskIntegrationGitHub'
+import TaskIntegrationGitLab from '../../../database/types/TaskIntegrationGitLab'
+import TaskIntegrationJira from '../../../database/types/TaskIntegrationJira'
+import TaskIntegrationJiraServer from '../../../database/types/TaskIntegrationJiraServer'
 import {DataLoaderWorker} from '../../graphql'
 
-export interface IntegrationByTeamId {
-  id: string
-  userId: string
-  service: 'github' | 'jira'
-  nameWithOwner: string | null
-  projectKey: string | null
-  projectName: string | null
-  avatar: string | null
-  cloudId: string | null
-  lastUsedAt: Date
-}
-
-type PrevJiraRepoIntegration = {
-  service: 'jira'
-  userId: string
-  issueKey: string
-  cloudId: string
-  nameWithOwner: null
+type TaskIntegration = (
+  | TaskIntegrationGitLab
+  | TaskIntegrationJira
+  | TaskIntegrationAzureDevOps
+  | TaskIntegrationGitHub
+  | TaskIntegrationJiraServer
+) & {
   lastUsedAt: Date
 }
 
-type PrevGitHubRepoIntegration = {
-  service: 'github'
-  userId: string
-  projectKey: null
-  cloudId: null
-  nameWithOwner: string
-  lastUsedAt: Date
-}
-
-type PrevRepoIntegrationRes = PrevGitHubRepoIntegration | PrevJiraRepoIntegration
-
 export const getPrevRepoIntegrations = async (
   userId: string,
   teamId: string,
   permLookup: Unpromise<ReturnType<typeof getPermsByTaskService>>
 ) => {
   const r = await getRethink()
-  const res = (await (
+  const prevIntegrationsRes = (await (
```

Comment:
Hey @mattkrick, yesterday during your office hours we discussed removing prev repo integrations from rethinkdb and storing them in Redis. 

You suggested that I check the speed of this nasty query in prod in Rethinkdb and see if it's slow. Here are the results:

- 277ms for me in the Product team
- 48ms for me in the Product Squad 3's team
- 334ms for you in the Product team
- 371ms for Jordan in the Product team
- 337 ms for Jordan in the Growth team

That doesn't seem that slow to me given that you and Jordan have probably created a lot more tasks than the average user.  

Do you think I should go ahead with the Redis refactor? If so, the plan for that was:

- when we call `repoIntegrations`, cache all the values in Redis
- yesterday I think we said that the key would be `teamId` + `integrationServiceName`, but I think we'd need to include the `userId` in that key too
- we would set a TTL of 3 months on the Redis key. Every time we fetch the key, we'd update the TTL
- the Redis values would be an array of objects which include all of the fields we get in `repoIntegrations`
- we would include a `lastUsedAt` field in the object which we would sort the results by

Reply:
Up to you! Are there other users that might have slower queries? Instead of targeting the average user wait time, could we say something like "99% of users have a latency under 500ms"? Do we cache on the client side so it's quicker after the first time I open the menu?

On the side, if we don't cache in redis is there a way to keep the query from growing in complexity with every new integration that we add? If we get rid of the query, could we also remove other fields like a label or display name kept in Task.integration? If the goal is to have a really fast initial fetch, is there a way to avoid reaching out to each integration to get the repo name?